tools/configuration-converter.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to convert istio configuration to sail operator configuration.
+# In the end of the execution new yaml file will be created with "sail-ISTIO_CONFIG_YAML" name.
+# Usage: ./configuration-converter.sh ISTIO_CONFIG_YAML_WITH_PATH, example:  ./configuration-converter.sh sample_config.yaml"
+set -e
+
+# Function to show usage
+usage() {
+    echo "Usage: $0 <input> [output] [-n <namespace>] [-v <version>]"
+    echo "  <input>         : Input file (required)"
+    echo "  [output]        : Output file (optional, defaults to input's base directory with '-sail.yaml' suffix)"
+    echo "  -n <namespace>  : Namespace (optional, defaults to 'istio-system')"
+    echo "  -v <version>    : Istio Version (optional)"
+    exit 1
+}
+
+# Ensure at least one file argument is provided
+if [[ $# -lt 1 ]]; then
+    usage
+fi
+
+# Initialize variables
+INPUT=""
+OUTPUT=""
+NAMESPACE="istio-system"
+VERSION=""
+
+# Identify the first and second file paths (treat the first file as INPUT)
+for arg in "$@"; do
+    if [[ -f "$arg" && -z "$INPUT" ]]; then
+        INPUT="$arg"  # First file is the INPUT
+    elif [[ -f "$arg" && -z "$OUTPUT" ]]; then
+        OUTPUT="$arg"  # Second file is the OUTPUT
+    fi
+done
+
+# Ensure the input file is provided and is valid
+if [[ -z "$INPUT" || ! -f "$INPUT" ]]; then
+    echo "Error: Input file is missing or invalid."
+    usage
+fi
+
+# If OUTPUT is not specified, generate a default output file name
+if [[ -z "$OUTPUT" ]]; then
+    OUTPUT="$(dirname "$INPUT")/$(basename "$INPUT" .yaml)-sail.yaml"
+fi
+
+# Ensure the output is not a directory
+if [[ -d "$OUTPUT" ]]; then
+    echo "Error: OUTPUT must be a file, not a directory."
+    exit 1
+fi
+
+# Parse optional namespace (-n) and version (-v) arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -n)
+            NAMESPACE="$2"
+            shift 2
+            ;;
+        -v)
+            VERSION="$2"
+            shift 2
+            ;;
+        *)
+            shift
+            ;;
+    esac
+done
+
+if ! command -v yq &>/dev/null; then
+    echo "Error: 'yq' is not installed. Please install it before running the script."
+    exit 1
+fi
+
+function add_mandatory_fields(){
+    # If VERSION is not empty, add .spec.version
+    if [[ -n "$VERSION" ]]; then
+        yq -i ".spec.version = \"$VERSION\"" "$OUTPUT"  
+    fi
```

Comment:
I'd move this to the bottom of this function. The most important part is that large yq command, so it makes sense for that to be the first thing we do. Adding the version is just an additional step. We should always try to keep the most important part of the function at the top, because it makes searching through the code faster (you immediately see the main point of a function without having to read the entire thing).